### PR TITLE
ci: Replace PAPR with CoreOS CI 

### PR DESCRIPTION
**Pull request by cgwalters**
Move the alternative builds into the Jenkinsfile.

Update it to do a FCOS build + kola run.

We drop the flatpak/rpm-ostree runs for now; the former
will needs some work to do the automatic virt bits.  The
latter I think we can circle back to when we e.g. figure
out how to include rpm-ostree's tests in kola runs.

### CODE REVIEW

**Comment by cgwalters** [comment]
/override papr-stub

Yes, a pass! :tada:

File: .cci.jenkinsfile [hunk 1]
```diff
@@ -34,3 +62,51 @@ codestyle: {
...
+  }
+},
+buildopts: {
+  coreos.pod(image: COSA_BUILDROOT_IMAGE, runAsUser: 0) {
+      checkout scm
+      sh """
+        set -xeuo pipefail
+        git submodule update --init
+
+        git clean -dfx && git submodule foreach git clean -dfx
```
Comment (jlebon) [inline .cci.jenkinsfile]: Hmm, we don't need this, right? Each pod should get a fresh workspace.

File: .papr.yml [hunk 2]
```diff
@@ -1,145 +1,9 @@
-# This suite skips the RPMs and does the build+unit tests in a container
+# Stub until we drop PAPR entirely
```
Comment (jlebon) [inline .papr.yml]: Mehh... let's just nuke it if we're not gonna leave any job in it? Once we drop PAPR here, I think it's just bubblewrap really that needs to migrate off before we can kill it entirely!

File: .cci.jenkinsfile [hunk 3]
```diff
@@ -5,8 +5,8 @@
 COSA_IMAGE = 'quay.io/coreos-assembler/coreos-assembler:latest'
 COSA_BUILDROOT_IMAGE = 'registry.svc.ci.openshift.org/coreos/cosa-buildroot:latest'
 
-stage("Build") {
-parallel rpms: {
+stage("Build and sanitychecks") {
```
Comment (jlebon) [inline .cci.jenkinsfile]: Bikeshed/optional: maybe just `Build` ? And then have the parallel branches be e.g. `normal`, `minimal`, and `codestyle`?

File: .cci.jenkinsfile [hunk 4]
```diff
@@ -15,8 +15,36 @@ parallel rpms: {
         git fetch origin --tags
         git submodule update --init
 
+        env SKIP_INSTALLDEPS=1 ./ci/build.sh
+      """
+      try {
+        sh """
+        make check
```
Comment (jlebon) [inline .cci.jenkinsfile]: I'd put this in a separate step, e.g. `Check` or something.
When tests fail, having more distinct and descriptive stages makes it easier to narrow down what went wrong I find. But yeah, totally good to just tweak this later too!

File: .cci.jenkinsfile (hunk 1 above)
Comment (cgwalters) [reply .cci.jenkinsfile]: Yeah, but I'm running multiple builds here.  I could split it into a separate pod but...some worry about the cost of repeated source cloning.

File: .papr.yml (hunk 2 above)
Comment (cgwalters) [reply .papr.yml]: I worried that papr would barf if there wasn't a `papr.yaml` until we removed the webhooks entirely?

Comment (jlebon) [reply .papr.yml]: It should be able to handle it gracefully.

File: .cci.jenkinsfile (hunk 1 above)
Comment (jlebon) [reply .cci.jenkinsfile]: I don't follow; this branch allocates its own pod already, no?

Comment (cgwalters) [reply .cci.jenkinsfile]: You're right that the first clean is unnecessary, but I was trying to be defensive when the section is copy-pasted.  However, I switched this now to use a `git worktree` which is actually nicer anyways.

Comment (jlebon) [reply .cci.jenkinsfile]: Ahh gotcha, I somehow missed the copy/paste pattern you were going for. But yeah, `git worktree` is cleaner.

**Pull request merged**